### ISSUE TITLE
StatefulSet can be orphaned after podTemplateSpec update

### ISSUE THREAD

**Issue by jsanda** [datastax/cass-operator#120]
**Summary**
If you update `.spec.podTemplateSpec` and then subsequently delete the `CassandraDatacenter`, the Statefulset does not get deleted. It looks like this is caused by the controller reference not getting set on the updated StatefulSet after `podTemplateSpec` is updated.

This is happening in master.

**Steps to Reproduce**
Deploy `CassandraDatacenter`:

```yaml
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
metadata:
  name: dc1
spec:
  clusterName: test-cluster
  serverType: cassandra
  serverVersion: 3.11.6
  managementApiAuth:
    insecure: {}
  size: 3
  storageConfig:
    cassandraDataVolumeClaimSpec:
      storageClassName: server
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 5Gi
``` 

After the `dc1` is ready, update the `podTemplateSpec`:

```yaml
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
metadata:
  name: dc1
spec:
  clusterName: test-cluster
  serverType: cassandra
  serverVersion: 3.11.6
  managementApiAuth:
    insecure: {}
  size: 3
  storageConfig:
    cassandraDataVolumeClaimSpec:
      storageClassName: server
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 5Gi
  podTemplateSpec:
    spec:
      containers: []
      terminationGracePeriodSeconds: 31
```

Delete `dc1` and there is no cascading delete for the StatefulSet,`test-cluster-dc1-default-sts`.

**Comment by jimdickinson** [comment]
Great find!

**Comment by jimdickinson** [comment]
Fixed by #121
